### PR TITLE
Provide additional explanation for checkboxes in PR template

### PR DESCRIPTION
**Pull request by Alexsandruss**
# UPDATED TEMPLATE

## Description

_Add a comprehensive description of proposed changes_

_List associated issue number(s) if exist(s): #6 (for example)_

_Documentation PR (if needed): #1340 (for example)_

_Benchmarks PR (if needed): https://github.com/IntelPython/scikit-learn_bench/pull/155 (for example)_

---

PR should start as a draft, then move to ready for review state after CI passed and checkboxes closed.
This approach ensures that reviewers don't spend extra time requesting changes for regular appearing things.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

File: .github/Pull_Request_template.md [hunk 1]
```diff
@@ -10,6 +10,12 @@ _Benchmarks PR (if needed): https://github.com/IntelPython/scikit-learn_bench/pu
 
 ---
 
+PR should start as a draft, then move to ready for review state after CI passed and checkboxes closed.
+This approach ensures that reviewers don't spend extra time requesting changes for regular appearing things.
```
Comment (david-cortes-intel) [inline .github/Pull_Request_template.md:14]: What does "regular appearing things" mean here?

File: .github/Pull_Request_template.md [hunk 2]
```diff
@@ -10,6 +10,12 @@ _Benchmarks PR (if needed): https://github.com/IntelPython/scikit-learn_bench/pu
 
 ---
 
+PR should start as a draft, then move to ready for review state after CI passed and checkboxes closed.
```
Comment (david-cortes-intel) [inline .github/Pull_Request_template.md:13]: Does "after CI passed" include the `/intelci: run` parts?

Comment (david-cortes-intel) [inline .github/Pull_Request_template.md:13]: Perhaps it should mention "after checkboxes are ticked or removed due to being inapplicable".

File: .github/Pull_Request_template.md (hunk 1 above)
Comment (Alexsandruss) [reply .github/Pull_Request_template.md:14]: - "Add comment explaining this hard-to-understand part"
- "Provide perf. measurements"
- "Update documentation to reflect changes"
- "Add test for this function"

File: .github/Pull_Request_template.md (hunk 2 above)
Comment (Alexsandruss) [reply .github/Pull_Request_template.md:13]: Since external users are not able to read intelci, no.

**Comment by Alexsandruss** [comment]
> A couple topics that would be good to cover here:
> 
> * What happens if the CI fails after changes that happened as a result of review suggestions and similar?

Requirements are for *moving from draft state* to ensure PR meets some readiness for review.

> * What happens if the CI failures are due to random sporadic issues?

`- [ ] All CI jobs are green or I have provided justification why they aren't.`

File: .github/Pull_Request_template.md (hunk 2 above)
Comment (Alexsandruss) [reply .github/Pull_Request_template.md]: Rephrased to `applicable checkboxes are closed`.

**Review by david-cortes-intel — APPROVED** [review]

**Pull request merged**